lib/AST/ASTImporterLookupTable.cpp

```diff
@@ -27,19 +27,37 @@ struct Builder : RecursiveASTVisitor<Builder> {
     LT.add(D);
     return true;
   }
+  // Friend declarations introduce new types, which should be visible during
+  // lookup of a specific named declaration even if it would be illegal to
+  // reference them at the root level of the TU.
+  // This visitor is necessary because in case of some friend declarations the
+  // AST does not contain the referenced friend class or function as a child of
+  // the referencing declaration, thus causing the recursive discovery fail.
+  // This visitor helps this discovery process by making the lookup table aware
+  // of these "hidden" declarations.
```

Comment:
The comments can be made shorter.
As I understand, this should mean that the FriendDecl node contains in part of the cases AST children with the "underlying" friend objects and these are visited, so no action is needed in the friend visitor. In the other cases the "befriended class" must be visited explicitly (add to table). The "dependent type" case belongs to the first case here when no action is needed.

Reply:
:+1: